core/src/main/scala/org/apache/spark/status/api/v1/JsonRootResource.scala

```diff
@@ -0,0 +1,148 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.status.api.v1
+
+import javax.servlet.ServletContext
+import javax.ws.rs._
+import javax.ws.rs.core.{Context, Response}
+
+import com.sun.jersey.api.core.ResourceConfig
+import com.sun.jersey.spi.container.servlet.ServletContainer
+import org.eclipse.jetty.server.handler.ContextHandler
+import org.eclipse.jetty.servlet.{ServletContextHandler, ServletHolder}
+
+import org.apache.spark.SecurityManager
+import org.apache.spark.ui.SparkUI
+
+@Path("/v1")
+private[v1] class JsonRootResource extends UIRootFromServletContext {
+
+  @Path("applications")
+  def getApplicationList(): ApplicationListResource = {
+    new ApplicationListResource(uiRoot)
+  }
+
+  @Path("applications/{appId}")
+  def getApplication(): OneApplicationResource = {
+    new OneApplicationResource(uiRoot)
+  }
+
+  @Path("applications/{appId}/jobs")
+  def getJobs(): AllJobsResource = {
+    new AllJobsResource(uiRoot)
+  }
+
+  @Path("applications/{appId}/jobs/{jobId: \\d+}")
+  def getJob(): OneJobResource = {
+    new OneJobResource(uiRoot)
+  }
+
+
+  @Path("applications/{appId}/executors")
+  def getExecutors(): ExecutorListResource = {
+    new ExecutorListResource(uiRoot)
+  }
+
+  @Path("applications/{appId}/stages")
+  def getStages(): AllStagesResource= {
+    new AllStagesResource(uiRoot)
+  }
+
+  @Path("applications/{appId}/stages/{stageId: \\d+}")
+  def getStage(): OneStageResource= {
+    new OneStageResource(uiRoot)
+  }
+
+  @Path("applications/{appId}/stages/{stageId: \\d+}/{attemptId: \\d+}")
+  def getStageAttempt(): OneStageAttemptResource= {
+    new OneStageAttemptResource(uiRoot)
+  }
+
+
+  @Path("applications/{appId}/storage/rdd")
+  def getRdds(): AllRDDResource = {
+    new AllRDDResource(uiRoot)
+  }
+
+  @Path("applications/{appId}/storage/rdd/{rddId: \\d+}")
+  def getRdd(): OneRDDResource = {
+    new OneRDDResource(uiRoot)
+  }
+
+}
+
+private[spark] object JsonRootResource {
+
+  def getJsonServlet(uiRoot: UIRoot): ServletContextHandler = {
+    val jerseyContext = new ServletContextHandler(ServletContextHandler.NO_SESSIONS)
+    jerseyContext.setContextPath("/json")
+    val holder:ServletHolder = new ServletHolder(classOf[ServletContainer])
+    holder.setInitParameter("com.sun.jersey.config.property.resourceConfigClass",
+      "com.sun.jersey.api.core.PackagesResourceConfig")
+    holder.setInitParameter("com.sun.jersey.config.property.packages",
+      "org.apache.spark.status.api.v1")
+    holder.setInitParameter(ResourceConfig.PROPERTY_CONTAINER_REQUEST_FILTERS,
+      classOf[SecurityFilter].getCanonicalName)
+    UIRootFromServletContext.setUiRoot(jerseyContext, uiRoot)
+    jerseyContext.addServlet(holder, "/*")
+    jerseyContext
+  }
+}
+
+
+private[spark] trait UIRoot {
```

Comment:
It might be nice to add an explanatory comment to this trait, e.g. to explain that it's inherited by both the Master and Application web UIs in order to abstract away such and such...